### PR TITLE
Enhance parser of vdo_status 

### PR DESCRIPTION
**Pull request by xhe123**
Enhance parser of vdo_status 

* Fix the "VDO statistics: not available" can not be parsed issue https://github.com/RedHatInsights/insights-core/issues/2209
* fix: #2209

Signed-off-by: Xiaonan He <xhe@redhat.com>

### CODE REVIEW

**Comment by xhe123** [comment]
code coverage is 100%.

```
(.python2) [xiaonan@dhcp-136-18 insights-core]$ pytest -svk vdo --cov=insights/parsers |grep vdo
insights/parsers/tests/test_vdo_status.py::test_vdo_status2 PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_exp2_0 PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_exp2_1 PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_simple PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_empty PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_full PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_documentation PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_exp PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_exp3 PASSED
insights/parsers/vdo_status.py                          41      0      4      0   100%
```

File: insights/parsers/vdo_status.py [hunk 1]
```diff
@@ -103,14 +105,26 @@ def __get_dev_mapper__(self, vol):
...
-            err_path = "['VDOs'][%s]['VDO statistics'][%s]" % (vol, dm_path)
-            raise KeyError('No key(s) named: %s' % err_path)
+            err_path = "['VDOs'][%s]" % (vol)
+            raise KeyError('No key(s) named: %s in %s' % (vol, err_path))
+
+        # Parse next layer of ['VDO statistics'] when vol exists.
+        vs = self.data['VDOs'][vol]['VDO statistics']
+
+        # No device mapper path when 'VDO statistics' value is a string of 'not available'
+        if isinstance(vs, str):
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: Better change to `if not isinstance(vs, dict):`

File: insights/parsers/vdo_status.py [hunk 2]
```diff
@@ -103,14 +105,26 @@ def __get_dev_mapper__(self, vol):
...
+            raise ParseException('Not available device mapper path in string \'%s\'' % vs)
+
+        # Has device mapper path when 'VDO statistics' value is a dict
+        dm = list(vs.keys())[0]
+        return vs[dm]
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: Better `return list(vs.keys())[0]`

File: insights/parsers/vdo_status.py (hunk 1 above)
Comment (xhe123) [reply insights/parsers/vdo_status.py]: yes, your suggestion is better.

File: insights/parsers/vdo_status.py (hunk 2 above)
Comment (xhe123) [reply insights/parsers/vdo_status.py]: yes, your suggestion looks better.

**Comment by xhe123** [comment]
@xiangce I improved code as you suggested, will you review again? Thanks!

**Pull request merged**